private/bufpkg/bufmodule/bufmodule.go

```diff
@@ -189,6 +193,13 @@ func ModuleWithModuleIdentityAndCommit(moduleIdentity bufmoduleref.ModuleIdentit
 	}
 }
 
+// ModuleWithID is used to construct a Module with an id.
```

Comment:
What is an ID on a Module? What is its form? What is this for? What's the motivation

Same general comments - I'd like us to be able to come back to the code in this PR in 6-12 months and be able to immediately understand the purpose of all of this without digging through PRs.